marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/entities/search/EntitySearchManager.java

```diff
@@ -246,18 +247,23 @@ public void exportRows(JsonNode queryDocument, Long limit, OutputStream out) {
         List<SearchQuery.SortOrder> sortOrder = searchQuery.getSortOrder().orElse(new ArrayList<>());
         ArrayNode sortOrderNode = sortOrderToArrayNode(sortOrder);
 
-
-        JsonNode opticPlanNode = EntitySearchService.on(finalDatabaseClient).getOpticPlan(structuredQuery, searchText, queryOptions, entityTypeId, entityTypeId, limit, sortOrderNode, columns.stream());
-        StringHandle stringHandle = new StringHandle(opticPlanNode.toString());
-        RowManager rowManager = finalDatabaseClient.newRowManager();
-        try (ReaderHandle readerHandle = new ReaderHandle()) {
-            rowManager.resultDoc(rowManager.newRawPlanDefinition(stringHandle), readerHandle.withMimetype(CSV_CONTENT_TYPE));
-            readerHandle.write(out);
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        } finally {
-            IOUtils.closeQuietly(out);
-        }
+        EntitySearchService entitySearchService = EntitySearchService.on(finalDatabaseClient);
+        // Exporting directly from Data Service to avoid bug https://bugtrack.marklogic.com/55338 related to namespaced path range indexes
+        Stream<Reader> export = entitySearchService.exportSearch(structuredQuery, searchText, queryOptions, entityTypeId, entityTypeId, limit, sortOrderNode, columns.stream(), CSV_CONTENT_TYPE);
```

Comment:
If the intent of this operation is to export search results as CSV, I think the method name should imply that - e.g. "exportSearchResultsAsCsv".

Reply:
It is passed the acceptType and the server then determines if it should be CSV. Right now CSV is the only format that will work, but I was trying to open that to be configurable. Similar to the REST rows endpoint.